frontend/src/seqvars/components/EffectsSelect/EffectsSelect.vue

```diff
@@ -0,0 +1,211 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+import {
+  SeqvarsQueryPresetsConsequence,
+  SeqvarsTranscriptTypeChoiceList,
+  SeqvarsVariantConsequenceChoiceList,
+  SeqvarsVariantTypeChoiceList,
+} from '@varfish-org/varfish-api/lib'
+
+import { Query } from '@/seqvars/types'
+
+import CheckButton from '../ui/CheckButton.vue'
+import CollapsibleGroup from '../ui/CollapsibleGroup.vue'
+import Input from '../ui/Input.vue'
+import PresetSelect from '../ui/PresetSelect.vue'
+
+import { matchesEffectsPreset } from './utils'
+
+const VARIANT_TYPES = {
+  snv: 'SNV',
+  indel: 'indel',
+  mnv: 'MNV',
+  complex_substitution: 'complex substitution',
+} satisfies Record<SeqvarsVariantTypeChoiceList[number], string>
+
+const TRANSCRIPT_TYPES = {
+  coding: 'coding',
+  non_coding: 'non-coding',
+} satisfies Record<SeqvarsTranscriptTypeChoiceList[number], string>
+
+const CUSTOMIZATION = {
+  Coding: {
+    disruptive_inframe_deletion: 'disruptive in-frame deletion',
+    disruptive_inframe_insertion: 'disruptive in-frame insertion',
+    // feature_truncation: 'feature truncation',
+    // frameshift_elongation: 'frameshift elongation',
+    frameshift_variant: 'frameshift variant',
+    // inframe_deletion: 'inframe deletion',
+    // inframe_insertion: 'inframe insertion',
+    // internal_elongation: 'internal elongation',
+    missense_variant: 'missense',
+    // mnv: 'MNV',
+    start_lost: 'start lost',
+    stop_gained: 'stop gained',
+    stop_retained_variant: 'stop retained',
+    stop_lost: 'stop lost',
+    // tandem_duplication: 'tandem duplication',
+  },
+  'Off-Exome': {
+    downstream_gene_variant: 'downstream',
+    intron_variant: 'intronic (coding transcript)',
+    // intergenic: 'intergenic',
+    upstream_gene_variant: 'upstream',
+    // exon_loss: 'exon loss',
+  },
+  'Non-coding': {
+    '3_prime_UTR_variant-exon_variant': "3' UTR exonic",
+    '3_prime_UTR_variant-intron_variant': "3' UTR intronic",
+    '5_prime_UTR_variant-exon_variant': "5' UTR exonic",
+    '5_prime_UTR_variant-intron_variant': "5' UTR intronic",
+    non_coding_transcript_exon_variant: 'non-coding exonic',
+    non_coding_transcript_intron_variant: 'non-coding intronic',
+  },
+  Splicing: {
+    splice_acceptor_variant: 'splice acceptor',
+    splice_donor_variant: 'splice donor',
+    splice_region_variant: 'splice region',
+  },
+  Structural: {
+    // structural: 'structural ',
+    // transcript_ablation: 'transcript ablation',
```

Comment:
@holtgrewe all the commented out entries are present in the design, but missing in the API.

Most of them seem to be present here:
https://github.com/varfish-org/varfish-server/blob/8d39e44c3d0d0265d353a5f4f17b52fd175755c8/backend/seqvars/models.py#L109-L208

Should I leave them out for now, or uncomment them on the backend side (+migrations etc.)?

Reply:
Please leave as is, this is (sadly) still work in progress.

Reply:
(wip on the server side / backend) - will fix and adjust later, possibly after our common project